src/Components/DatePicker/Component.purs

```diff
@@ -241,6 +241,13 @@ setSelection selection = do
   H.raise $ SelectionChanged selection
   synchronize
 
+setSelectionWithoutRaising :: forall m. MonadAff m => Maybe Date -> CompositeComponentM m Unit
+setSelectionWithoutRaising selection = do
+  st <- H.get
+  let targetDate = maybe st.targetDate (\d -> (year d) /\ (month d)) selection
+  H.modify_ _ { selection = selection, targetDate = targetDate }
+  synchronize
+
```

Comment:
I'm weary of the potential for the `setSelection` logic having to change and the engineer not realizing they should also change `setSelectionWithoutRaising`.

What do you think of updating `setSelection` to something like:

```purs
setSelection :: forall m. MonadAff m => Maybe Date -> CompositeComponentM m Unit
setSelection selection = do
  setSelectionWithoutRaising selection
  H.raise (SelectionChanged selection)
```

I don't _think_ the order of when `synchronize` is called is important here. I believe that just updates the calendar view to match the selected date, so it seems fine to call it before raising.

Reply:
Makes sense to me. It seems like `synchronize` just messes around with the state, so I agree that it probably doesn't matter if it's called before or after the raise. Honestly, it feels like it should all happen before the raise anyway, and likely it's a hold-over from the halogen 5 conversion. But, I didn't look into it that closely.